src/v3/src/components/LoopbackProbe/LoopbackProbe.tsx

```diff
@@ -93,14 +95,29 @@ const LoopbackProbe: FunctionComponent<{ uischema: LoopbackProbeElement }> = ({
     challengeRequest,
   } = deviceChallengePayload;
 
-  const cancelHandler = (params?: ActionParams) => {
-    if (typeof cancelStep !== 'undefined') {
-      onSubmitHandler({
-        isActionStep: true,
-        step: cancelStep,
+  const submitHandler = async (stepName: string) => {
+    const payload: IdxActionParams = {
+      step: stepName,
+    };
+    if (typeof idxTransaction?.context.stateHandle !== 'undefined') {
+      payload.stateHandle = idxTransaction.context.stateHandle;
+    }
+    const newTransaction = await authClient?.idx.proceed(payload);
+    setIdxTransaction(newTransaction);
+  };
+
+  const cancelHandler = async (params?: ActionParams) => {
+    const payload: IdxActionParams = {
+      actions: [{
+        name: cancelStep,
         params,
-      });
+      }],
+    };
+    if (typeof idxTransaction?.context.stateHandle !== 'undefined') {
+      payload.stateHandle = idxTransaction.context.stateHandle;
     }
+    const newTransaction = await authClient?.idx.proceed(payload);
+    setIdxTransaction(newTransaction);
```

Comment:
I considered refactoring these into one method but the payloads differ in a way that I think is a bit messy to try to reconcile, and the remaining duplication is minimal so I'm keeping them separate.

Reply:
Yea, I totally agree with this, the action payload vs the standard step + params payload is a mess to deal with, totally good with how you have it.